Makefile

```diff
@@ -106,8 +106,8 @@ $(GOFUMPT): $(TOOLS_DIR)/go.mod
 
 gofumpt: $(GOFUMPT) ## Build gofumpt
 
-$(GOLANGCI_LINT): $(TOOLS_DIR)/go.mod
-	cd $(TOOLS_DIR); go mod download; go build -tags=tools -o bin/golangci-lint github.com/golangci/golangci-lint/cmd/golangci-lint
+$(GOLANGCI_LINT): ## Get the golangci-lint binary to match gh workflow version instead of building for consistency
+	cd $(TOOLS_DIR); curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(TOOLS_BIN_DIR) v1.55.2
```

Comment:
this will have the same problem of drifting away from the workflow, won't it? you've just moved where the dupe hardcoded version is

Reply:
I actually ran into a few problems when trying to figure out this linting issue:
The linting results were drastically different from workflow even with the linting config.

So I downgrading hack/tools golangci-lint to 1.55 from 1.60 but then running make lint produced some missing dependency even after go mod tidy. 
Then i just downloaded the binary for 1.55.2 in this command - which actually worked and produced the same results as the workflow

I can try upgrading both, but with 1.60 i noticed that the //nolint wasn't working for our c file references in our go code and i havent figured that out. And we would need to update our linter config since some have been deprecated

Reply:
nvm, upgraded version, set cgo=0 and updated linter config to remove deprecated value resolved the mismatch

Reply:
Not sure if it's happening in this instance, but @snguyen64 you should know that golangci-lint is sensitive to the version of Go that it was built with. Weird things have happened in the past where golangci-lint was built with a different version of Go than the module itself.